main.py

```diff
@@ -3695,10 +3695,58 @@ def do_command(self, command, args, sender, channel, tags, isWhisper=False):
                     config["promoschanged"] = "yes"
                     cur.execute("REPLACE INTO config(name, value) VALUES('promoschanged', 'yes')")
                 return
+            if command == "freepacks" or (command == "bet" and len(args) > 0 and args[0].lower() == "packs"):
+                if len(args) > 0 and args[0].lower() in ["open", "claim", "redeem"]:
+                    if len(args) < 2:
+                        self.message(channel, "Usage: !freepacks open <booster name>", isWhisper)
+                        return
+                    with db.cursor() as cur:
+                        cur.execute("SELECT remaining, boostername FROM freepacks WHERE userid = %s AND boostername = %s", [tags['user-id'], args[1]])
+                        result = cur.fetchone()
+
+                        if result is None or result[0] == 0:
+                            self.message(channel, "You don't have any free packs of that type left to claim!", isWhisper)
+                            return
+
+                        # can they actually open it?
+                        cur.execute("SELECT COUNT(*) FROM boosters_opened WHERE userid = %s AND status = 'open'",
```

Comment:
I know this is a feature update, but the [wisdom of StackOverflow](https://stackoverflow.com/a/5264695) says that an `EXISTS` subquery [could be faster up to 25%](https://stackoverflow.com/a/5264757). Given how big `boosters_opened` will be getting since nobody ever seems to delete from there (and things depend on that!), this may be worthwhile to introduce as a pattern in the entire codebase.